File: backend/src/main/java/org/example/backend/modules/edu/entity/ProjectGroupApplication.java
package org.example.backend.modules.edu.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.time.LocalDateTime;

@Data
@TableName("project_group_application")
public class ProjectGroupApplication {

    @TableId(type = IdType.ASSIGN_ID)
    private Long id;

    private Long batchId;

    private Long topicId;

    private Long leaderId;

    private String groupName;

    private String projectName;

    private String projectDescription;

    private String repoUrl;

    private String deployUrl;

    private String applyReason;

    private String status;

    private Long reviewerId;

    private String reviewComment;

    private LocalDateTime reviewTime;

    private Long generatedGroupId;

    private LocalDateTime createTime;

    private LocalDateTime updateTime;
}

File: frontend/src/api/projectGroupApplication.ts
import request from '@/utils/request'

export interface ProjectGroupApplicationItem {
  id: string
  batchId: string
  topicId: string
  leaderId: string
  groupName: string
  projectName: string | null
  projectDescription: string | null
  repoUrl: string | null
  deployUrl: string | null
  applyReason: string | null
  status: 'PENDING' | 'APPROVED' | 'REJECTED' | 'CANCELED' | string
  reviewerId: string | null
  reviewComment: string | null
  reviewTime: string | null
  generatedGroupId: string | null
  createTime: string | null
  updateTime: string | null
}

export interface ProjectGroupApplicationPageQuery {
  current: number
  size: number
  groupName?: string
  batchId?: string
  topicId?: string
  leaderId?: string
  status?: string
}

export interface ProjectGroupApplicationCreateFormData {
  batchId: string
  topicId: string
  groupName: string
  projectName: string
  projectDescription: string
  repoUrl: string
  deployUrl: string
  applyReason: string
  memberIds: string[]
}

export interface ProjectGroupApplicationReviewFormData {
  id?: string
  status: 'APPROVED' | 'REJECTED'
  reviewComment: string
}

export interface PageResult<T> {
  records: T[]
  total: number
  size: number
  current: number
  pages: number
}

export const getProjectGroupApplicationPageApi = (params: ProjectGroupApplicationPageQuery) => {
  return request.get<any, { code: number; message: string; data: PageResult<ProjectGroupApplicationItem> }>(
    '/api/project-group-applications/page',
    {
      params,
    },
  )
}

export const getProjectGroupApplicationDetailApi = (id: string) => {
  return request.get<any, { code: number; message: string; data: ProjectGroupApplicationItem }>(
    `/api/project-group-applications/${id}`,
  )
}

export const createProjectGroupApplicationApi = (data: ProjectGroupApplicationCreateFormData) => {
  return request.post<any, { code: number; message: string; data: ProjectGroupApplicationItem }>(
    '/api/project-group-applications',
    data,
  )
}

export const reviewProjectGroupApplicationApi = (data: ProjectGroupApplicationReviewFormData) => {
  return request.put<any, { code: number; message: string; data: ProjectGroupApplicationItem }>(
    '/api/project-group-applications/review',
    data,
  )
}

export const cancelProjectGroupApplicationApi = (id: string) => {
  return request.put<any, { code: number; message: string; data: ProjectGroupApplicationItem }>(
    `/api/project-group-applications/${id}/cancel`,
  )
}